### PR TITLE
VideoPlayer: add setting for double/triple buffers

### PR DESCRIPTION
**Pull request by afedchin**
see title

@FernetMenta ping

### CODE REVIEW

**Comment by zag2me** [comment]
Better String:
"Defines the number of presentation buffers that are used by the graphics driver. Select 2 if the driver uses double buffering or 3 for triple buffering"

**Comment by afedchin** [comment]
@zag2me thanks.
updated

**Comment by FernetMenta** [comment]
+1
thanks for continuing work on this

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
@afedchin could you rebase and resolve the merge conflict please

**Comment by afedchin** [comment]
will do asap

**Comment by afedchin** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
Does this work on all platforms? If not this setting needs to be hidden on those.

**Comment by afedchin** [comment]
this is base implementation. maintainers of those platforms may hide setting if it's not needed.

**Comment by FernetMenta** [comment]
@MartijnKaijser your question does not match 100% to the problem :)
For a/v sync VideoPlayer needs to know display latency. This is dependent on the number of buffers used be the driver. Prior to this patch we assumed double buffering but in the meantime many systems have triple buffers. Unfortunately this is not detectable on many systems, hence the setting. If a platform or appliance knows the number, it can hide the setting from the user.

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -18444,7 +18447,13 @@ msgctxt "#36551"
 msgid "Defines the strength of the stereoscopic 3D effect in the GUI. This is done by controlling the depth of perception within the GUI, so the higher the value, the more elements will pop out of the screen. [Zero] Disables the stereoscopic 3D effect of the GUI.[CR]For a good visual experience, the value should be higher for small screens and lower for large screens. Note: this is not supported by all skins."
 msgstr ""
 
-#empty strings from id 36552 to 36599
+#. Description of setting "System -> Video output -> NoOfBuffers" with label #36043
+#: system/settings/settings.xml
+msgctxt "#36552"
+msgid "Defines the number of presentation buffers used by the graphics driver. Select 2 if the driver uses double buffering or 3 for triple buffering."
```
Comment (FernetMenta) [inline addons/resource.language.en_gb/resources/strings.po]: "defines" sounds misleading to me. This value does not affect the number of buffers used by the driver.

Comment (afedchin) [inline addons/resource.language.en_gb/resources/strings.po]: any suggestions?

**Pull request merged**